shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm

```diff
@@ -247,17 +255,26 @@ - (BOOL)accessibilityPerformMagicTap {
     // TODO(abarth): Use extract once we're at C++17.
   }
 
-  ReleaseObjects(doomed_objects);
-  doomed_objects.clear();
-
-  if (root) {
-    if (!view_.accessibilityElements) {
-      view_.accessibilityElements = @[ root ];
+  SemanticsObject* doomed_focused_object = nil;
+  for (const auto& entry : doomed_objects) {
+    SemanticsObject* object = entry.second;
+    if ([object accessibilityElementIsFocused]) {
+      doomed_focused_object = object;
+      break;
     }
+  }
+
+  if (doomed_focused_object != nil) {
+    // Previously focused element is no longer in the tree, let iOS figure out what to focus next.
+    // TODO(goderbauer): Figure out which element should be focused next and post
+    //     UIAccessibilityLayoutChangedNotification with that element instead.
+    UIAccessibilityPostNotification(UIAccessibilityScreenChangedNotification, nil);
   } else {
-    view_.accessibilityElements = nil;
+    UIAccessibilityPostNotification(UIAccessibilityLayoutChangedNotification, nil);
   }
-  UIAccessibilityPostNotification(UIAccessibilityLayoutChangedNotification, nil);
+
+  ReleaseObjects(doomed_objects);
+  doomed_objects.clear();
```

Comment:
Maybe do this in `ReleaseObjects` and kill this line here and in the dtor? I can't think of a situation where you'd want a set of freed object pointers.